lib/elixir/test/elixir/module/types/pattern_test.exs

```diff
@@ -76,8 +90,14 @@ defmodule Module.Types.PatternTest do
       assert quoted_pattern(false) == {:ok, {:atom, false}}
       assert quoted_pattern(:foo) == {:ok, {:atom, :foo}}
       assert quoted_pattern(0) == {:ok, :integer}
-      assert quoted_pattern(0.0) == {:ok, :float}
+      assert quoted_pattern(+0.0) == {:ok, :float}
+      assert quoted_pattern(-0.0) == {:ok, :float}
       assert quoted_pattern("foo") == {:ok, :binary}
+
+      assert {{:ok, :float}, [diagnostic]} = quoted_pattern_with_diagnostics(0.0)
+
+      assert diagnostic.message =~
+               "pattern matching on 0.0 is equivalent to matching only on +0.0"
```

Comment:
I struggled way too much to disable this warning 😅
It is also on https://github.com/elixir-lang/elixir/pull/12950 I believe